optin/reporting/test/utils/Validation.spec.js

```diff
@@ -0,0 +1,373 @@
+	describe('unsupported validator type', () => {
+		it('should return error given validator type not supported', () => {
+			// Arrange:
+			const method = 'notSupported';
+
+			// Act: + Assert:
+			expect(() => Validation.validate(method)).to.throw(`${method} is not supported`);
+		});
+	});
+
+	describe('getCompleted query', () => {
+		// Arrange:
+		const validator = Validation.validate('getCompleted');
+
+		describe('paginationValidation', () => {
+			runBasicPaginationValidationTests(validator);
+		});
+
+		describe('sortValidation', () => {
+			// Arrange:
+			const invalidParams = [123, 'invalid'];
+			const validParams = ['asc', 'desc', 'none', ''];
+
+			runBasicSortDirectionValidationTests(invalidParams, validParams, validator);
+		});
+
+		describe('addressTransactionHashValidation', () => {
+			// Arrange:
+			const invalidParams = [123, 'invalid'];
+			const validNemAddress = ['NA5UJVHGHUXUFA2VFQOJAUEXZIHPU3FUK7GRDUBM', ''];
+			const validTxHash = ['ba9232776840ffb6adbf116820f2341aed3ca26b316b118844ccb050fe73bac8', ''];
+
+			runBasicNemAddressValidationTests(invalidParams, validNemAddress, validator);
+			runBasicTransactionHashValidationTests(invalidParams, validTxHash, validator);
+		});
+
+		describe('sortBy', () => {
+			// Arrange:
+			const invalidParams = ['invalid', 123];
+			const validSortBy = ['nemHashes', 'symbolHashes'];
+
+			runBasicSortByValidationTests(invalidParams, validSortBy, validator);
+		});
+
+		describe('optinType', () => {
+			// Arrange:
+			const queryField = 'optinType';
+
+			it('should return error given incorrect optinType', async () => {
+				// Arrange:
+				const invalidParams = ['invalid'];
+
+				await runInvalidAsserts({
+					validator,
+					invalidParams,
+					queryField,
+					errorMessage: 'optin type must be pre or post'
+				});
+			});
+
+			it('should return 0 error given correct sortBy', async () => {
+				// Arrange:
+				const validParams = ['pre', 'post'];
+
+				await runValidAsserts({
+					validator,
+					validParams,
+					queryField
+				});
+			});
+		});
+
+		describe('symbolAddress', () => {
+			// Arrange:
+			const queryField = 'symbolAddress';
+
+			it('should return error given incorrect symbolAddress', async () => {
+				// Arrange:
+				const invalidParams = ['NDCWGCUSOWJBD3JKOQOIWACPWMCVA6LVAWYPC3PI'];
+
+				await runInvalidAsserts({
+					validator,
+					invalidParams,
+					queryField,
+					errorMessage: 'symbol address must be 39 character long'
+				});
+			});
+
+			it('should return 0 error given correct symbolAddress', async () => {
+				// Arrange:
+				const validParams = ['NAIVQSJ6IU2NCDWWZSUYKKXK7JTGROW6FDRQTJY', 'NC6PLXOJLS43WIH23CV7OZGXTWQI3QNXNLBA7MY'];
+
+				await runValidAsserts({
+					validator,
+					validParams,
+					queryField
+				});
+			});
+		});
+	});
+
+	describe('getOptinRequests query', () => {
+		const validator = Validation.validate('getOptinRequests');
+
+		describe('paginationValidation', () => {
```

Comment:
consider naming like
describe 'getOptinRequests'
describe 'pagination validator'
it 'succeeds when all parameters are valid'
OR if 'fails when any parameter is omitted'

and try to improve other test names to so they form sentences :) 


	describe('getOptinRequests query', () => {
		const validator = Validation.validate('getOptinRequests');

		describe('paginationValidation', () => {